Lib/concurrent/futures/thread.py

```diff
@@ -148,4 +150,5 @@ def shutdown(self, wait=True):
         if wait:
             for t in self._threads:
                 t.join()
+        self._threads.clear()
```

Comment:
@pitrou: I have the feeling you're getting obsessed about "dangling threads".

That's true. In my experience, sometimes, it shows a larger issue. For example, https://bugs.python.org/issue31249 pointed a reference cycle which wasn't seen before.